pkg/users/users.js

```diff
@@ -188,66 +177,76 @@ async function getLogins(logind_client) {
                 ["org.freedesktop.login1.User", "State"],
                 { type: "ss", flags: "", timeout: 5000 });
             if (active.v !== "closing")
-                currentLogins.push(name);
+                details[name] = { ...details[name], loggedIn: true };
         }));
     } catch (err) {
         console.warn("Unexpected error when getting logged in accounts", err);
     }
 
+    // locked password
+
     // shadow-utils passwd supports an --all flag which is lacking on RHEL and
     // stable Fedora releases. Available at least on Fedora since
     // shadow-utils-4.14.0-5.fc40 (currently known as rawhide).
-    const locked_users_map = {};
     try {
         const locked_statuses = await cockpit.spawn(["passwd", "-S", "--all"], { superuser: "require", err: "message", environ: ["LC_ALL=C"] });
         // Slice off the last empty line
         for (const line of locked_statuses.trim().split('\n')) {
-            const username = line.split(" ")[0];
+            const name = line.split(" ")[0];
             const status = line.split(" ")[1];
-            locked_users_map[username] = status == "L";
+            details[name] = { ...details[name], isLocked: status === "L" };
         }
     } catch (err) {
-        // Only warn when it is unrelated to --all.
-        if (err.message && !err.message.includes("bad argument --all")) {
+        if (err.message?.includes("bad argument --all")) {
+            // Fallback for old passwd
+            try {
+                const shadow = await cockpit.file("/etc/shadow", { superuser: "require", err: "message" }).read();
+                for (const line of shadow.split('\n')) {
+                    const [name, hash] = line.split(":");
+                    if (name && hash)
+                        details[name] = { ...details[name], isLocked: hash.startsWith("!") };
```

Comment:
These 6 added lines are not executed by any test.

Reply:
This is because fedora 40 has `--all`. But it is tested on e.g. rhel-9-5 or ubuntu-2204.